packages/astro/src/runtime/server/transition.ts

```diff
@@ -65,6 +74,20 @@ export function renderTransition(
 	return scope;
 }
 
+export function createAnimationScope(
+	transitionName: string,
+	animations: Record<string, TransitionAnimationPair>
+) {
+	const hash = Math.random().toString(36).slice(2, 8);
+	const scope = `astro-${hash}`;
```

Comment:
How can I create a good identifier for the scope?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID would be ideal but i don't think we can depend on that yet. I think the way you did it is fine for now.